src/main/java/weavers/siltarae/login/domain/TokenProvider.java

```diff
@@ -0,0 +1,127 @@
+        this.secretKey = Keys.hmacShaKeyFor(secretKey.getBytes(StandardCharsets.UTF_8));
+        this.accessTokenExpirationTime = accessTokenExpirationTime;
+        this.refreshTokenExpirationTime = refreshTokenExpirationTime;
+        this.refreshTokenRepository = refreshTokenRepository;
+    }
+
+    public TokenPair createTokenPair(final Long memberId) {
+        String accessToken = createAccessToken(memberId.toString(), accessTokenExpirationTime);
+        String refreshToken = createRefreshToken();
+
+        saveRefreshToken(refreshToken, memberId);
+
+        return TokenPair.builder()
+                .accessToken(accessToken)
+                .refreshToken(refreshToken)
+                .refreshTokenExpirationTime(refreshTokenExpirationTime)
+                .build();
+    }
+
+    private String createAccessToken(final String subject, final Long expirationTime) {
+        final Date now = new Date();
+        final Date expirationDate = new Date(now.getTime()+expirationTime);
+
+        return Jwts.builder()
+                .subject(subject)
+                .issuedAt(now)
+                .expiration(expirationDate)
+                .signWith(secretKey)
+                .compact();
+    }
+
+    private String createRefreshToken() {
+        return UUID.randomUUID().toString();
+    }
+
+    public String renewAccessToken(final Long memberId) {
+        return createAccessToken(memberId.toString(), accessTokenExpirationTime);
+    }
+
+    private void saveRefreshToken(final String refreshToken, final Long memberId) {
+        RefreshToken createdRefreshToken = RefreshToken.builder()
+                .memberId(memberId)
+                .refreshToken(refreshToken)
+                .build();
+
+        refreshTokenRepository.save(createdRefreshToken);
+    }
+
+    public boolean isExpiredAccessAndValidRefresh(final String accessToken, final String refreshToken) {
+        try {
+            Long memberId = getMemberIdFromAccessToken(accessToken);
+            validateRefreshToken(refreshToken, memberId);
+        } catch (RefreshTokenException | BadRequestException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isValidTokens(final String accessToken, final String refreshToken) {
+        try {
+            Long memberId = getMemberIdFromAccessToken(accessToken);
+            validateRefreshToken(refreshToken, memberId);
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    private void validateRefreshToken(final String token, final Long memberId) {
+        RefreshToken refreshToken = refreshTokenRepository.findById(memberId)
+                .orElseThrow(() -> new RefreshTokenException(ExceptionCode.EXPIRED_REFRESH_TOKEN));
+
+        if(!refreshToken.getRefreshToken().equals(token)) {
+            throw new BadRequestException(ExceptionCode.INVALID_REFRESH_TOKEN);
+        }
```

Comment:
가독성 측면에서 개인적인 의견인데 isTokenMismatch 라는 메소드를 통해서 true/false 체크를 하는 것도 좋을 것 같습니다~

Reply:
```
private void validateRefreshToken(final String token, final Long memberId) {
    RefreshToken refreshToken = refreshTokenRepository.findById(memberId)
            .orElseThrow(() -> new RefreshTokenException(ExceptionCode.EXPIRED_REFRESH_TOKEN));

    if (isTokenMismatch(token, refreshToken)) {
        throw new BadRequestException(ExceptionCode.INVALID_REFRESH_TOKEN);
    }
}

private boolean isTokenMismatch(String token, RefreshToken refreshToken) {
    return !refreshToken.getRefreshToken().equals(token);
}
```

이렇게 수정해보면 어떨까요?

Reply:
해당 부분 고민했는데 이미 TokenProvider의 메서드가 충분히 많은 상황이라 따로 분리하지 않는 쪽이 낫겠다고 생각했습니다! if문 내 부정형이 걱정된다면 아래와 같은 형태는 어떨까요?

```suggestion
        if(refreshToken.getRefreshToken().equals(token)) return;
        throw new BadRequestException(ExceptionCode.INVALID_REFRESH_TOKEN);
```

Reply:
좋아요! 👍